be/src/starrocks_format/starrocks_lib.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file is based on code available under the Apache license here:
+//   https://github.com/apache/incubator-doris/blob/master/be/src/service/doris_main.cpp
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+#include <aws/core/Aws.h>
+#include <aws/core/client/ClientConfiguration.h>
+#include <glog/logging.h>
+#include <unistd.h>
+
+#include <filesystem>
+#include <fstream>
+
+#include "common/config.h"
+#include "fs/fs_s3.h"
+#include "runtime/time_types.h"
+#include "storage/lake/fixed_location_provider.h"
+#include "storage/lake/tablet_manager.h"
+#include "storage/olap_define.h"
+#include "util/timezone_utils.h"
+
+namespace starrocks::lake {
+
+static bool _starrocks_format_inited = false;
+Aws::SDKOptions aws_sdk_options;
+
+lake::TabletManager* _lake_tablet_manager = nullptr;
+
+void starrocks_format_initialize(void) {
+    setenv("STARROCKS_HOME", "./", 0);
+    setenv("UDF_RUNTIME_DIR", "./", 0);
+
+    if (!_starrocks_format_inited) {
```

Comment:
this is not safe in multi-thread env, either use mutex or leverage std::call_once.

Reply:
This object does not need to be initialized multiple times, nor does it need to be shared across multiple threads.

Reply:
since `Aws::InitAPI(aws_sdk_options);` is called in this function, and it is process wide effected, multi-init may cause unexpected result.